faunadb-android/src/test/java/com/faunadb/client/AndroidClientSpec.java

```diff
@@ -0,0 +1,116 @@
+package com.faunadb.client;
+
+import com.faunadb.client.dsl.DslSpec;
+import com.faunadb.client.errors.BadRequestException;
+import com.faunadb.client.errors.UnauthorizedException;
+import com.faunadb.client.query.Expr;
+import com.faunadb.client.types.Value;
+import com.google.common.collect.ImmutableList;
+import com.google.common.util.concurrent.ListenableFuture;
+import org.junit.AfterClass;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+import java.util.List;
+
+import static com.faunadb.client.database.SetupDatabase.createDatabase;
+import static com.faunadb.client.database.SetupDatabase.createServerKey;
+import static com.faunadb.client.database.SetupDatabase.dropDatabase;
+import static com.faunadb.client.query.Language.*;
+import static com.faunadb.client.types.Codec.BOOLEAN;
+import static com.faunadb.client.types.Codec.STRING;
+import static com.faunadb.client.types.Value.NullV.NULL;
+import static com.google.common.base.Functions.constant;
+import static com.google.common.util.concurrent.Futures.*;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.core.Is.is;
+import static org.hamcrest.core.Is.isA;
+
+public class AndroidClientSpec extends DslSpec {
+
+  private static final String DB_NAME = "faunadb-java-android-test";
+  private static final Expr DB_REF = Ref("databases/" + DB_NAME);
+
+  private static FaunaClient rootClient;
+  private static FaunaClient client;
+
+  @BeforeClass
+  public static void setUpClient() throws Exception {
+    rootClient = createFaunaClient(ROOT_TOKEN);
+
+    catching(rootClient.query(dropDatabase(DB_REF)), BadRequestException.class, constant(NULL)).get();
+    Value db = rootClient.query(createDatabase(DB_NAME)).get();
+    Value key = rootClient.query(createServerKey(db)).get();
+
+    client = createClientWithServerKey(key);
+  }
+
+  @AfterClass
+  public static void closeClients() throws Exception {
+    catching(rootClient.query(dropDatabase(DB_REF)), BadRequestException.class, constant(NULL)).get();
```

Comment:
It might be a good idea to add one, though.
People using OkHttpClient have the ability to close the underlying resources if they need to: http://square.github.io/okhttp/3.x/okhttp/

By wrapping the http connection, we limit the user ability to shutdown the http connection if he needs to.